crates/router/src/connector/braintree/transformers.rs

```diff
@@ -255,234 +544,1268 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
 #[serde(rename_all = "camelCase")]
-pub struct TransactionResponse {
-    id: String,
-    currency_iso_code: String,
-    amount: String,
-    status: BraintreePaymentStatus,
+pub struct DataResponse {
+    charge_credit_card: AuthChargeCreditCard,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Default, Debug, Clone, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct BraintreeApiErrorResponse {
-    pub api_error_response: ApiErrorResponse,
+pub struct RefundInputData {
+    amount: StringMajorUnit,
+    merchant_account_id: Secret<String>,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
-pub struct ErrorsObject {
-    pub errors: Vec<ErrorObject>,
-    pub transaction: Option<TransactionError>,
+#[derive(Default, Debug, Clone, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BraintreeRefundInput {
+    transaction_id: String,
+    refund: RefundInputData,
 }
-#[derive(Debug, Deserialize, Serialize)]
+
+#[derive(Default, Debug, Clone, Serialize)]
+pub struct BraintreeRefundVariables {
+    input: BraintreeRefundInput,
+}
+
+#[derive(Default, Debug, Clone, Serialize)]
+pub struct BraintreeRefundRequest {
+    query: String,
+    variables: BraintreeRefundVariables,
+}
+
+impl<F> TryFrom<BraintreeRouterData<&types::RefundsRouterData<F>>> for BraintreeRefundRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: BraintreeRouterData<&types::RefundsRouterData<F>>,
+    ) -> Result<Self, Self::Error> {
+        let metadata: BraintreeMeta =
+            utils::to_connector_meta_from_secret(item.router_data.connector_meta_data.clone())
+                .change_context(errors::ConnectorError::InvalidConnectorConfig {
+                    config: "metadata",
+                })?;
+
+        utils::validate_currency(
+            item.router_data.request.currency,
+            Some(metadata.merchant_config_currency),
+        )?;
+        let query = REFUND_TRANSACTION_MUTATION.to_string();
+        let variables = BraintreeRefundVariables {
+            input: BraintreeRefundInput {
+                transaction_id: item.router_data.request.connector_transaction_id.clone(),
+                refund: RefundInputData {
+                    amount: item.amount,
+                    merchant_account_id: metadata.merchant_account_id,
+                },
+            },
+        };
+        Ok(Self { query, variables })
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, strum::Display)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum BraintreeRefundStatus {
+    SettlementPending,
+    Settling,
+    Settled,
+    SubmittedForSettlement,
+    Failed,
+}
+
+impl From<BraintreeRefundStatus> for enums::RefundStatus {
+    fn from(item: BraintreeRefundStatus) -> Self {
+        match item {
+            BraintreeRefundStatus::Settled | BraintreeRefundStatus::Settling => Self::Success,
+            BraintreeRefundStatus::SubmittedForSettlement
```

Comment:
Can `SubmittedForSettlement` be considered as success?

Reply:
No , they actually have some status like this 
```
    SettlementPending,
    SettlementDeclined,
    SettlementConfirmed,
```